### PR TITLE
adding charter document for the BoF meeting

### PR DESCRIPTION
**Pull request by dimmyvi**

### CODE REVIEW

File: charter.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+There are many situations in which it is desirable to share a digital credential with another person. For example, you may want to provide access to your vehicle to a Valet. You may also want to provide access to your home to your cat sitter. Or, you may want to share a hotel key with your spouse. Today, no such standardized method exists in a cross-platform, multidisciplinary capacity. 
+
+The WG charter includes the definition and standardization of a protocol that will facilitate such credential transfers from individual to individual. The protocol will leverage a “relay server” to transfer data from sender to recipient. The scope of the transfer is limited to a single origin device and a single destination device.
+
+Privacy goals include:
+
+- The relay server should not see sensitive details of the share
+- The relay server should not be able to provision the credential itself, acting as an intermediary for the recipient (MiTM)
+- The relay server should not know the identity of the sender nor receiver
+- Device OEMs should not be able to identity the social graph of whom has shared with whom
```
Comment (alexpell00) [inline charter.md:12]: I don't think this is needed. If the point above it is upheld then the relay server wouldn't be able to build a social graph, and our internal privacy concerns and mitigations seen unrelated to the EITF.

File: charter.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Sufficient security measures should be embedded in the protocol in an effort to:
+
+- Ensure only the intended recipient is able to provision the credential
+- Ensure the credential can only be provisioned once (Anti-replay)
+- Ensure the sender has intent to share (secure user intent)
```
Comment (alexpell00) [inline charter.md:18]: The protocol doesn't enforce anything around user intent. I think we can remove this point.

File: charter.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+- Allow the sender to revoke the credential, either before or after the recipient has accepted, where applicable
```
Comment (alexpell00) [inline charter.md:19]:
```suggestion
- Allow only the sender and receiver to mutate the sharing channel once a connection has been established.
```

This seems more related to share management and is outside the scope of the IETF spec. I think we could reword this to only cover mailbox deletion.

File: charter.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+The solution the WG comes up with must:
+
+- Allow a sender to initiate a share
+- Allow a recipient to view the share request, and provision the credential associated with the share upon receipt
```
Comment (alexpell00) [inline charter.md:24]: I think we should break this out a bit more. Maybe something like:

- Allow a recipient to view an open graph preview of a share
- Allow dynamic message formats based on the credential type
- Allow sender device and receiver device to quickly perform multiple round trip communications

File: charter.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+There are many situations in which it is desirable to share a digital credential with another person. For example, you may want to provide access to your vehicle to a Valet. You may also want to provide access to your home to your cat sitter. Or, you may want to share a hotel key with your spouse. Today, no such standardized method exists in a cross-platform, multidisciplinary capacity. 
+
+The WG charter includes the definition and standardization of a protocol that will facilitate such credential transfers from individual to individual. The protocol will leverage a “relay server” to transfer data from sender to recipient. The scope of the transfer is limited to a single origin device and a single destination device.
+
+Privacy goals include:
+
+- The relay server should not see sensitive details of the share
+- The relay server should not be able to provision the credential itself, acting as an intermediary for the recipient (MiTM)
+- The relay server should not know the identity of the sender nor receiver
```
Comment (crystalyq) [inline charter.md:11]: I don't think this rule always applies to the sender. Because the sender client and relay server are both managed by the same organization, where the identity is already known in the ecosystem in most cases. The spec should allow proprietary oauth built on top of this spec for sender and relay communication

File: charter.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+- Device OEMs should not be able to identity the social graph of whom has shared with whom
+
```
Comment (crystalyq) [inline charter.md:13]: also
- The relay server should only retain the share info for a limited amount of time

File: charter.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+The solution the WG comes up with must:
+
+- Allow a sender to initiate a share
+- Allow a recipient to view the share request, and provision the credential associated with the share upon receipt
+
+Out of scope topics for the proposed WG are:
+
+- Defining the mechanism the receiver will use (which APIs) in order to provision the credential
+- The WG will define the full set of different credential types that could be shared. A subset of these credential types adhere to a public standard. For these credential types, the format of the Provisioning Information shall be defined by the appropriate standard. Other credential types may be proprietary. For these credential types, the protocol the WG aims to establish shall not define the actual format nor content of each field within the Provisioning Information.
```
Comment (crystalyq) [inline charter.md:29]: For other credential types, should we add that they adhere to the proprietary specs of the credential issuing authorities?

File: charter.md (hunk 3 above)
Comment (nicksha) [reply charter.md:19]: +1 
termination of the credential is specific to the standards it follows, anything outside of the mailbox management should be out of scope. Also, deleting the mailbox may not result in successfully terminating a credential. Example with car keys, deleting the mailbox after sender uploads the key import request will not guarantee terminating the shared car key.

File: charter.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+#Charter for Working Group
+
+There are many situations in which it is desirable to share a digital credential with another person. For example, you may want to provide access to your vehicle to a Valet. You may also want to provide access to your home to your cat sitter. Or, you may want to share a hotel key with your spouse. Today, no such standardized method exists in a cross-platform, multidisciplinary capacity. 
+
+The WG charter includes the definition and standardization of a protocol that will facilitate such credential transfers from individual to individual. The protocol will leverage a “relay server” to transfer data from sender to recipient. The scope of the transfer is limited to a single origin device and a single destination device.
+
+Privacy goals include:
+
+- The relay server should not see sensitive details of the share
+- The relay server should not be able to provision the credential itself, acting as an intermediary for the recipient (MiTM)
```
Comment (ykarandikar) [inline charter.md:10]: To use more inclusive language I would suggest AttackerInTheMiddle instead of ManInTheMiddle

File: charter.md (hunk 1 above)
Comment (nicksha) [reply charter.md:12]: +1

File: charter.md (hunk 5 above)
Comment (ykarandikar) [reply charter.md:11]: The spirit of this goal is to make sure that the relay server doesn't try to know identities of sender and receiver. 
In theory sender client and relay server can be managed by separate organizations. 
Even in case where they are managed by same org the spirit of the goal should apply.

File: charter.md (hunk 3 above)
Comment (ykarandikar) [reply charter.md:19]: +1

Role of relay ends after the sharing flow is over (either with success or failure).

Removing the credential should fall back to lifecycle management  policies of the vertical.

File: charter.md (hunk 4 above)
Comment (ykarandikar) [reply charter.md:24]: I would refrain from mentioning use of open graph for viewing in the charter document. Specific technologies can change in future.

File: charter.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+- Defining the mechanism the receiver will use (which APIs) in order to provision the credential
+- The WG will define the full set of different credential types that could be shared. A subset of these credential types adhere to a public standard. For these credential types, the format of the Provisioning Information shall be defined by the appropriate standard. Other credential types may be proprietary. For these credential types, the protocol the WG aims to establish shall not define the actual format nor content of each field within the Provisioning Information.
+- The User Interface (UI) that is displayed to the sender or receiver during sending or receiving - this will depend on the device OEM’s UI and HI guidelines.
+
+The WG will deliver a set of APIs and workflows to facilitate secure credential transfer. The WG must consider all Privacy and Security considerations in an effort to perform the credential transfer in a secure manner. Privacy-preserving algorithms such as field-level encryption will be used to protect data in transit.
+
+The starting point for Secure Credential Transfer WG discussions shall be draft-secure-credential-transfer.
```
Comment (ykarandikar) [inline charter.md:34]: I'm a bit confused by bullet points that say "WG will" in Out of scope topics.

Does it mean that WG will not do any of the things mentioned here or it will?

File: charter.md (hunk 8 above)
Comment (dimmyvi) [reply charter.md:10]: IMHO, ManInTheMiddle (MITM) attack is a wide-recognized term in security. If WG feels it needs to be changed to more inclusive, we'll do it, but we shall start as it is to avoid confusion (security world is rather conservative and keeps using common vocabulary)

File: charter.md (hunk 5 above)
Comment (dimmyvi) [reply charter.md:11]: I believe Crystal's comment is valid - we may want to change wording from "should not know the identity" to something like "should not store/track the identity"

File: charter.md (hunk 1 above)
Comment (dimmyvi) [reply charter.md:12]: Agreed

File: charter.md (hunk 2 above)
Comment (dimmyvi) [reply charter.md:18]: I believe we want to support the ability to ensure secure user intent from sender - in a form of FIDO challenge/ device attestation. It is worth mentioning that in this charter doc

File: charter.md (hunk 4 above)
Comment (dimmyvi) [reply charter.md:24]: Agree with Yogesh here - the description should be as generic as possible, the concrete details to be defined in the draft itself.

File: charter.md (hunk 7 above)
Comment (dimmyvi) [reply charter.md:29]: Then it might require references to such specs, which is going to be challenging; thus, we believe, an opaque description used above is more appropriate.
Please feel free to add what you see it should look like here...

File: charter.md (hunk 9 above)
Comment (dimmyvi) [reply charter.md:34]: The last 2 paragraphs are a conclusion, rather than a part of "Out of scope" topic.
Sorry for confusion.

File: charter.md (hunk 6 above)
Comment (dimmyvi) [reply charter.md:13]: Valid point!

File: charter.md (hunk 7 above)
Comment (ykarandikar) [reply charter.md:29]: This bullet point under out of scope says "WG will define the full set of different credential types..."

It might be good to maybe reword that.

File: charter.md (hunk 4 above)
Comment (alexpell00) [reply charter.md:24]: Makes sense. We can drop the specifics about how the preview is shown.

File: charter.md (hunk 3 above)
Comment (dimmyvi) [inline charter.md:19]: removing the whole bullet point

File: charter.md (hunk 5 above)
Comment (dimmyvi) [reply charter.md:11]: changing the wording to "relay server should not persist the identity of the sender nor receiver"

File: charter.md (hunk 2 above)
Comment (dimmyvi) [reply charter.md:18]: let's keep the statement for now and wait for WG feedback
